Sources/KavaPlugin+AnalyticsPluginProtocol.swift

```diff
@@ -0,0 +1,218 @@
+// ===================================================================================================
+// Copyright (C) 2017 Kaltura Inc.
+//
+// Licensed under the AGPLv3 license, unless a different license for a
+// particular library is specified in the applicable library path.
+//
+// You may obtain a copy of the License at
+// https://www.gnu.org/licenses/agpl-3.0.html
+// ===================================================================================================
+
+
+import PlayKit
+
+extension KavaPlugin: AnalyticsPluginProtocol {
+    
+    /************************************************************/
+    // MARK: - AnalyticsPluginProtocol
+    /************************************************************/
+    
+    public var isFirstPlay: Bool {
+        get {
+            return self.kavaData.isFirstPlay
+        }
+        set(newValue) {
+            self.kavaData.isFirstPlay = newValue
+        }
+    }
+    
+    public var playerEventsToRegister: [PlayerEvent.Type] {
+        return [
+            PlayerEvent.stateChanged,
+            PlayerEvent.loadedMetadata,
+            PlayerEvent.play,
+            PlayerEvent.pause,
+            PlayerEvent.playing,
+            PlayerEvent.seeking,
+            PlayerEvent.sourceSelected,
+            PlayerEvent.ended,
+            PlayerEvent.playbackInfo,
+            PlayerEvent.trackChanged,
+            PlayerEvent.error
+        ]
+    }
+    
+    public func registerEvents() {
+        PKLog.debug("register player events")
+        
+        self.playerEventsToRegister.forEach { event in
```

Comment:
I implement `AnalyticsPluginProtocol`, in all other stats plugins it looks same. 
It makes sense since every event has his own callback implementation so we are going over relevant events list and add a specific callback for each event.